scripts/common.py

```diff
@@ -888,3 +891,21 @@ def setup_chrome_bin_env_variable() -> None:
     else:
         print('Chrome is not found, stopping...')
         raise Exception('Chrome not found.')
+
+
+def run_ng_compilation() -> None:
+    """Runs angular compilation."""
+    max_tries = 2
+    ng_bundles_dir_name = 'dist/oppia-angular'
+    for _ in range(max_tries):
+        try:
+            with servers.managed_ng_build() as proc:
+                proc.wait()
+        except subprocess.CalledProcessError as error:
+            print(error.output)
+            sys.exit(error.returncode)
+        if os.path.isdir(ng_bundles_dir_name):
+            break
```

Comment:
Tbh, I don't think the angular build fails flakely (I don't know if that is a word). For the same input source if consistently builds or fails. Maybe it can run into memory issue but I don't think that will be the case as webpack compiles a different bundle for each page. For the exception and retry logic, I saw this structure in webpack build and decided to keep it.